### PR TITLE
Improve working with large CRLs from OpenSSL

### PR DESCRIPTION
**Pull request by beldmit**
Fixes: #25603

##### Checklist

- [x] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

**Comment by beldmit** [comment]
This PR consists of 2 commits: documentation change and behavior change.

I think that it make sense to land the documentation change to all supported branches and either provide some big limit for CRL size to master branch and document it. What do you think about the limit (my commit leaves it unlimited)?

File: crypto/x509/x_all.c [hunk 1]
```diff
@@ -116,11 +116,12 @@ static ASN1_VALUE *simple_get_asn1(const char *url, BIO *bio, BIO *rbio,
                                    int timeout, const ASN1_ITEM *it)
 {
 #ifndef OPENSSL_NO_HTTP
+    size_t max_resp_len = (it == ASN1_ITEM_rptr(X509_CRL)) ? 0 : OSSL_HTTP_DEFAULT_MAX_RESP_LEN;
```
[the remark this answers is not in the document]
Comment (beldmit) [reply crypto/x509/x_all.c:119]: @t8m I agree - what value do you think reasonable?

[the remark this answers is not in the document]
Comment (t8m) [reply crypto/x509/x_all.c:119]: Let's say 16 MiB?
https://chasersystems.com/blog/an-analysis-of-certificate-revocation-list-sizes/
The largest has 12 MB.
So maybe 32 MiB to have some extra room?

[the remark this answers is not in the document]
Comment (beldmit) [reply crypto/x509/x_all.c]: Done

**Comment by beldmit** [comment]
I think that tests can be exempted and it's up to y to decide whether this change goes to all branches or only to master branch (and #25609 to the rest ones)

**Review by t8m — APPROVED** [review]
OK for all branches.

**Review by mattcaswell — APPROVED** [review]

**Comment by t8m** [comment]
@mattcaswell OK for stable branches?

**Comment by mattcaswell** [comment]
Yes

**Review by tom-cosgrove-arm — APPROVED** [review]
LGTM

**Comment by t8m** [comment]
Merged to all the active branches. Thank you for your contribution.

**Pull request closed without merging**